tests/test_routing.py

```diff
@@ -429,3 +429,24 @@ def handler2():
     _, handler, params = router.get("/path/to/bar", "three")
     assert handler() == "handler2"
     assert params == {"foo": "bar"}
+
+
+def test_identical_path_routes_with_different_methods(handler):
+    def handler1():
+        return "handler1"
+
+    def handler2():
+        return "handler2"
+
+    router = Router()
+    router.add("/<foo:path>", handler1, methods=["GET", "OPTIONS"])
+    router.add("/<foo:path>", handler2, methods=["POST"])
+    router.finalize()
+
+    _, handler, params = router.get("/a-random-path", "GET")
```

Comment:
Since we're doing path, can you make the test `/a/random/path`? Or perhaps even better, maybe parametrize the test and do both a single and multi segment path.

Reply:
@ahopkins Just pushed an update. Let me know if it works for you.